modules/video/test/ocl/test_optflowpyrlk.cpp

```diff
@@ -121,13 +124,24 @@ OCL_TEST_P(PyrLKOpticalFlow, Mat)
             float errdiff = 0.0f;
 
             if (!eq || errdiff > 1e-1)
+            {
                 ++mistmatch;
+                continue;
+            }
+
+            eq = std::abs(cpuErr[i] - err[i]) < 0.01;
+            if(!eq &&
+               winSize.width >= 16 && winSize.height >=16 &&
+               winSize.width <=24 && winSize.height <=24)
```

Comment:
> This test not only works for OpenCL, but also works for the CUDA and any other optimized version code.

CUDA has own API.
Available code path for this function is: CPU, CPU optimized with intrinsic (SSE/NEON), OpenCL, OpenVX (disabled due accuracy).

> How about that I add a new test specifically for OpenCL

No, you should not do this. At least you can't force OpenCL code path, especially if there is no OpenCL device or something is not supported (like `ocl::Image2D::isFormatSupported(CV_32F, 1, false)`).
This test is intended to check function output. It is launched on many platforms and we check that result is valid on all of them.

> In this kernel, the input winSize bigger than 24*24 will be referred as 24*24, the input winSize smaller than 16*16 will be referred as 16*16.

If OpenCL kernel doesn't support some `winSize` values (or other parameters)), then "host" code should check for these cases and bailout to other implementations.